lib/chef/knife/azurerm_server_create.rb

```diff
@@ -270,27 +270,27 @@ def ssh_override_winrm
       def set_default_image_reference!
         begin
           if locate_config_value(:azure_image_os_type)
-            if (locate_config_value(:azure_image_reference_publisher) || locate_config_value(:azure_image_reference_offer) || locate_config_value(:azure_image_reference_sku))
-              # if azure_image_os_type is given and other image reference parameters are also given,
+            if (locate_config_value(:azure_image_reference_publisher) || locate_config_value(:azure_image_reference_offer))
+              # if azure_image_os_type is given and any of the other image reference parameters like publisher or offer are also given,
               # raise error
-              raise ArgumentError, 'Please specify either --azure-image-os-type OR other image reference parameters i.e.
+              raise ArgumentError, 'Please specify either --azure-image-os-type OR --azure-image-os-type with --azure-image-reference-sku or 4 image reference parameters i.e.
                 --azure-image-reference-publisher, --azure-image-reference-offer, --azure-image-reference-sku, --azure-image-reference-version."'
```

Comment:
@adamedx , any suggestions on this error message? 
I think we should either simplify the error message OR just keep it the same way as it was `Please specify either --azure-image-os-type OR other image reference parameters` and explain the usage in the Readme.